src/RestrictedTokenFaucet.sol

```diff
@@ -0,0 +1,59 @@
+pragma solidity >=0.5.0;
+
+interface ERC20Like {
+    function balanceOf(address) external view returns (uint256);
+    function transfer(address,uint256) external; // return bool?
+}
+
+contract RestrictedTokenFaucet {
+    // --- Auth ---
+    mapping (address => uint) public wards;
+    function rely(address guy) public isOwner { wards[guy] = 1; }
+    function deny(address guy) public isOwner { wards[guy] = 0; }
+    modifier auth { require(wards[msg.sender] == 1); _; }
+    modifier isOwner { require(owner == msg.sender); _; }
+    address public owner;
+
+    uint256 public amt;
```

Comment:
wondering if it's weird to have the same `amt` for every `gem`? Like some tokens are worth a lot more than others. Maybe better to have an `amts` mapping? Maybe it doesn't matter for the FKR deployment?

Reply:
I think the idea of a multi-token faucet is a bit weird anyway. Do we actually use them this way?

Reply:
I tend to agree on the idea of having it be multitoken is a bit weird. From the deployment scripts, it seems to me to only be used for kovan MKR.  

I would be happy to change it to take a token as the constructor, but that will make it less hotswap-able with the current Faucet in the deployment scripts.  While this is a little weird, I would prefer to keep the Restricted contract as close to the regular contract as possible. 

@gbalabasquer can you confirm if we need a multi-token faucet? If we don't need that, then I would like update this to accept a token in the constructor and change the regular `token-faucet` and deployment scripts to match.